manta-js/examples/extension-example/src/App.tsx

```diff
@@ -261,13 +281,53 @@ export default function App() {
     return injected.privateWallet.subscribeWalletState(setStateInfo);
   }, [injected, setStateInfo]);
 
+  // sub accounts && unSub accounts
+  useEffect(() => {
+    if (!injected || !injected.privateWallet) {
+      return;
+    }
+    return injected.accounts.subscribe(async (accounts) => {
+      if (accounts.length <= 0) {
+        setPublicAddress('');
+        setZkAddress('');
+        localStorage.removeItem('connected');
+        return;
+      }
+      setPublicAddress(accounts[0].address);
+      // @ts-ignore
+      setZkAddress(accounts[0].zkAddress);
+      // @ts-ignore
+      setNetwork(accounts[0].network);
+      localStorage.setItem('connected', '1');
+    });
+  }, [setPublicAddress, setZkAddress, setNetwork, injected]);
+
+  useEffect(() => {
+    if (!injected || !injected.privateWallet || !currentNetwork) {
+      return;
+    }
+    const updateProvider = async () => {
+      const rpcUrl = currentNetwork.rpc instanceof Array
+        ? currentNetwork.rpc[(Math.random() * currentNetwork.rpc.length) | 0]
+        : currentNetwork.rpc;
+      const provider = new HttpProvider(rpcUrl);
+      // const provider = new WsProvider(rpcUrl);
+      setApi(null);
```

Comment:
Don't fully understand why this line is necessary, but probably a react thing I don't know about

Reply:
`setApi(null)` ? 

Right, some of the following states need to know that the api has changed